packages/logger/src/middleware/index.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { RequestHandler } from 'express';
+import ExpressRequestLogger from 'express-requests-logger';
+import { LogLevel, LogLevels, LoggerType } from '../client/index.js';
+
+import { MiddlewareData } from './dependencyDataTypes.js';
+
+export type ExpressLoggerConfig = {
+	logger: LoggerType;
+	excludeURLs?: string[];
+};
+
+/**
+ * We want to log at an appropriate level for the error code returned.
+ * System errors are logged as ERROR.
+ * Auth errors from the user are logged as WARN, this includes Forbidden and Unauthorized requests.
+ * Everything else is logged as INFO
+ * @param maybeResponse
+ * @returns
+ */
+const selectLevelForResponse = (maybeResponse: { response?: { status_code?: unknown } }): LogLevel => {
+	const statusCode = maybeResponse?.response?.status_code;
+
+	if (statusCode === undefined || typeof statusCode !== 'number') {
+		return LogLevels.INFO;
+	}
+
+	if (statusCode >= 500) {
+		return LogLevels.ERROR;
+	}
+
+	const authErrorCodes = [401, 403];
+	if (authErrorCodes.includes(statusCode)) {
+		return LogLevels.WARN;
+	}
+
+	return LogLevels.INFO;
+};
+
+// Disbaling eslint unused variable - We are passed params from the middleware but we don't use them. Leaving them in the function signature unused, just for awareness.
```

Comment:
"Disbaling" to "Disabling"

Reply:
Dang it I saw this and meant to change it :p updated